src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -0,0 +1,473 @@
+                case SyntaxKind.NamespaceDeclaration:
+                    return ((NamespaceDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.UsingDirective:
+                    return ((UsingDirectiveSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.ExternAliasDirective:
+                    return ((ExternAliasDirectiveSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.ClassDeclaration:
+                    return ((ClassDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.StructDeclaration:
+                    return ((StructDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.InterfaceDeclaration:
+                    return ((InterfaceDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.EnumDeclaration:
+                    return ((EnumDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.DelegateDeclaration:
+                    return ((DelegateDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.EventFieldDeclaration:
+                    return ((EventFieldDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.MethodDeclaration:
+                    return ((MethodDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.OperatorDeclaration:
+                    return ((OperatorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.ConversionOperatorDeclaration:
+                    return ((ConversionOperatorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.ConstructorDeclaration:
+                    return ((ConstructorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    return ((ConstructorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.DestructorDeclaration:
+                    return ((DestructorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.PropertyDeclaration:
+                    return ((PropertyDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.IndexerDeclaration:
+                    return ((IndexerDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                case SyntaxKind.AddAccessorDeclaration:
+                    return ((AccessorDeclarationSyntax)currentNode).SemicolonToken.IsMissing;
+                default:
+                    return false;
+            }
+        }
+
+        private static bool StatementClosingDelimiterExists(SyntaxNode currentNode, ref int lastDelimiterPosition)
```

Comment:
💡 `ref TextSpan` might be better, so you don't have to document whether the value here is the start or end position.